modules/security/src/main/java/org/eclipse/xpanse/modules/security/Oauth2WebSecurityFilter.java

```diff
@@ -149,4 +138,58 @@ CorsConfigurationSource corsConfigurationSource() {
     }
 
 
+    /**
+     * Configuration applied when method security is disabled.
+     */
+    @Profile("oauth")
```

Comment:
@baixinsui this is a topic that must work with "noauth" as well. 
Basically the idea is to allow xpanse to run completely without any authentication and authorization because in most of cases, the auth process happens outside the xpanse application runtime. So in this case, the role logic must also be removed. 
In case of running xpanse on HuaweiCloud, the API-GW will do the authentication (and authorization -> to be checked) for us. If API-GW does not support authorization (role check) then we will do it in the xpanse application in a custom way -> Read IAM to get roles assigned to user and then decide what methods to allow. This would mean that the default spring MVC roles extractor would not work.  

You can also discuss with WangLi once. He knows the background of this requirement.

Reply:
@swaroopar What you mean is when the property `enable.role.protection` is false, the  authentication should be disable also? In this case, maybe we need to rename the property to `enable.oauth.security` is more better.

Reply:
if authentication is enabled -> then authorization can be enabled or disabled
if authentication is disabled -> then authorization is by default disabled

Reply:
@swaroopar Disabling authentication and authorization is completed with the PR https://github.com/eclipse-xpanse/xpanse/pull/1499. This PR is not need to modify.

Reply:
Updated the code based on the PR https://github.com/eclipse-xpanse/xpanse/pull/1499